lib/types/ipblock.go

```diff
@@ -0,0 +1,173 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"net"
+	"strings"
+)
+
+// ipBlock represents a continuous segment of IP addresses
+type ipBlock struct {
+	// TODO rename count as it is actually the accumulation of the counts up to and including this
+	// on in the pool
+	// maybe add another field, although technically we need count only to find out when to enter this
+	// block not for anything else
+	firstIP, count *big.Int
```

Comment:
This was very confusing, since in the context of `ipBlock` you're using count like a count, but in the context of the `IPPool`, you're completely changing its meaning... I'd suggest you simply have 2 different types, an `ipBlock` with the current defintion and a new `ipPoolBlock` with a `firstIP` and `totalCount` or something like that (`lastIP`?). 

You can still reuse the `count` variable from the `ipBlock`, since you won't need it for anything else, so that you won't loose any performance. But it will be much cleaner semantically to have a different for it.

Reply:
I am going to wait for @imiric to finish his review as well before starting. But do you mean that you want to make ipBlock and than instead of using it in hte Pool to get it's elements and copy it to ipPoolBlock which has `count` named something different but still with the same value that it has now?

I propose `countWithThisBlock`  or `countUntilThisBlock` which will mean I can drop the if in the `GetIPBig` function

Reply:
> get it's elements and copy it to ipPoolBlock which has count named something different but still with the same value that it has now
yes, exactly that

> I propose countWithThisBlock or countUntilThisBlock
:+1: for both (the second one with changing the implementation slightly, of course)

Reply:
https://github.com/loadimpact/k6/pull/1682/commits/ebd5b470e182c2a868d6b91e0fa999aa12973b20